hopr/hopr-lib/src/lib.rs

```diff
@@ -570,8 +571,10 @@ impl Hopr {
     fn error_if_not_in_state(&self, state: HoprState, error: String) -> errors::Result<()> {
         if self.status() == state {
             Ok(())
+        } else if state == HoprState::Running {
+            Err(errors::HoprLibError::StatusError(HoprStatusError::NotRunningError))
         } else {
-            Err(errors::HoprLibError::StatusError(error))
+            Err(errors::HoprLibError::StatusError(HoprStatusError::General(error)))
         }
     }
```

Comment:
Why not return an `Err(HoprStatus::NotThereYet(HoprState))`

Reply:
It is used to check if the node is in `Uninitialized` state when starting.